tests/rptest/tests/redpanda_oauth_test.py

```diff
@@ -0,0 +1,150 @@
+import json
+
+from rptest.clients.python_librdkafka import PythonLibrdkafka
+from rptest.clients.types import TopicSpec
+from rptest.services.redpanda import LoggingConfig, RedpandaService, SecurityConfig, make_redpanda_service
+from rptest.services.keycloak import KeycloakService
+from rptest.services.cluster import cluster
+
+from ducktape.tests.test import Test
+from rptest.services.redpanda import make_redpanda_service
+from rptest.clients.rpk import RpkTool
+from rptest.util import expect_exception
+
+from confluent_kafka import KafkaException
+
+CLIENT_ID = 'myapp'
+
+
+class RedpandaOIDCTestBase(Test):
+    """
+    Base class for tests that use the Redpanda service with OIDC
+    """
+    def __init__(self,
+                 test_context,
+                 num_nodes=5,
+                 sasl_mechanisms=['SCRAM', 'OAUTHBEARER'],
+                 **kwargs):
+        super(RedpandaOIDCTestBase, self).__init__(test_context, **kwargs)
+        self.produce_messages = []
+        self.produce_errors = []
+        num_brokers = num_nodes - 1
+        self.keycloak = KeycloakService(test_context)
+
+        security = SecurityConfig()
+        security.enable_sasl = True
+        security.sasl_mechanisms = sasl_mechanisms
+
+        self.redpanda = make_redpanda_service(test_context, num_brokers)
+        self.redpanda.set_security_settings(security)
+
+        self.rpk = RpkTool(self.redpanda,
+                           username='admin',
+                           password='admin',
+                           sasl_mechanism="SCRAM-SHA-256")
+
+    def delivery_report(self, err, msg):
+        """
+            Reports the failure or success of a message delivery.
+            Successfully delivered messages are placed in `messages`.
+            Errors are placed in `errors`
+
+            Args:
+            err (KafkaError): The error that occurred on None on success.
+
+            msg (Message): The message that was produced or failed.
+
+        """
+        if err is not None:
+            self.produce_errors.append(
+                'Delivery failed for User record {}: {}'.format(
+                    msg.key(), err))
+            return
+        self.produce_messages.append(
+            'User record {} successfully produced to {} [{}] at offset {}'.
+            format(msg.key(), msg.topic(), msg.partition(), msg.offset()))
+
+    def setUp(self):
+        self.produce_messages.clear()
+        self.produce_errors.clear()
+        self.redpanda.logger.info("Starting Redpanda")
+        self.redpanda.start()
+
+
+class RedpandaOIDCTest(RedpandaOIDCTestBase):
+    @cluster(num_nodes=5)
+    def test_init(self):
+        kc_node = self.keycloak.nodes[0]
+        try:
+            self.keycloak.start_node(kc_node)
+        except Exception as e:
+            self.logger.error(f"{e}")
+            self.keycloak.clean_node(kc_node)
+            assert False, "Keycloak failed to start"
+
+        self.rpk.create_topic('foo')
```

Comment:
You'll also need to create an ACL for the principal over this topic. The default mapping from the token to a principal is that the principal is extracted from the `sub` claim of the access token (until that becomes more configurable).

In this case that might be `service-account-{CLIENT_ID}`, depending on how Keycloak works.